src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
```

Comment:
Don't we also want to clear the bit on the loResult? Presumably any overflow/carry from the low 32 bits will will get consumed in the upper 32 bits computation.